### ISSUE TITLE
Bug: MD030 (spaces after list markers) false positive on hard-wrapped brackets

### ISSUE THREAD

**Issue by chrisgrieser** [rvben/rumdl#253]
This bug was introduced in of the recent updates, not sure which.

Conditions for the issue to occur:
1. List item
2. At the end of the line, there is a parenthesis (e.g., when citing a source)
3. the last word in the parenthesis is on the next line due to the hard-wrapping.

```md
- foobar foobar foobar foobar foobar foobar foobar foobar foobar (Doe 2003, p.
  1234)
```

<img width="1169" height="352" alt="Image" src="https://github.com/user-attachments/assets/5bff146c-1a96-4b3c-aae6-df9c1b3cf1e5" />

---

Related to this issue, `rumdl fmt` does not alter the content of the notes with such false positives, but lsp-formatting does. lsp-formatting results in a trailing space being added, triggering MD009. The next lsp-format removes the trailing space, going back to triggering MD030
